Parse/src/main/java/com/parse/ParseOkHttpClient.java

```diff
@@ -146,9 +146,7 @@ public ParseOkHttpClient(int socketOperationTimeout, SSLSessionCache sslSessionC
     // Set Body
     ParseHttpBody parseBody = parseRequest.getBody();
     ParseOkHttpRequestBody okHttpRequestBody = null;
-    if(parseBody instanceof ParseByteArrayHttpBody) {
-      okHttpRequestBody = new ParseOkHttpRequestBody(parseBody);
-    }
+    okHttpRequestBody = new ParseOkHttpRequestBody(parseBody);
```

Comment:
Ideally, I do not think it will cause `NPE` sicne we only use `okHttpRequestBody ` for POST and PUT request. In that situation `parseBody ` should not be `null`. 
If we add the null checking, when the POST or PUT has a `null` body,  `okHttpRequestBuilder` will throw `IllegalArgumentException`.
If we do not add the null checking, when the POST or PUT has a `null` body, the exception will be thrown when we actually write the body.
In this situation, I think probably it is better to add the null checking. Let's prevent the error as early as possible.